scripts/src/java/org/oppia/android/scripts/maven/GenerateLicenseTexts.kt

```diff
@@ -0,0 +1,423 @@
+ *      resources corrsponding to each third-party dependency.
+ * - path_to_pb_file: relative path to textproto file that stores the list of dependencies
+ *      and their license links.
+ * Example:
+ *   bazel run //scripts:generate_license_texts -- $(pwd)/app/src/main/res/values
+ *   third_party_dependency_names.xml third_party_dependency_versions.xml
+ *   third_party_dependency_license_texts.xml third_party_dependency_license_texts_array.xml
+ *   third_party_dependency_license_names_array.xml scripts/assets/maven_dependencies.pb
+ */
+fun main(args: Array<String>) {
+  if (args.size < 7) {
+    println(
+      """
+      Usage: bazel run //scripts:generate_license_texts -- <path_to_directory_values>
+      <name_of_value_resource_file_1> <name_of_value_resource_file_2> 
+      <name_of_value_resource_file_2> <name_of_value_resource_file_4>
+      <name_of_value_resource_file_2> <path_to_pb_file>  
+      """.trimIndent()
+    )
+    throw Exception("Too less arguments passed.")
+  }
+  val pathToValuesDirectory = args[0]
+  val pathToNamesXml = "$pathToValuesDirectory/${args[1]}"
+  val pathToVersionsXml = "$pathToValuesDirectory/${args[2]}"
+  val pathToLicensesTextsXml = "$pathToValuesDirectory/${args[3]}"
+  val pathToLicenseTextArrayXml = "$pathToValuesDirectory/${args[4]}"
+  val pathToLicenseNamesArrayXml = "$pathToValuesDirectory/${args[5]}"
+  val pathToPbFile = args[6]
+
+  val mavenDependencyList = retrieveMavenDependencyList(pathToPbFile)
+  if (mavenDependencyList.isEmpty()) {
+    throw Exception(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE)
+  }
+  val copyrightLicenseSet = retrieveAllLicensesSet(mavenDependencyList)
+
+  writeDependenciesNamesXml(pathToNamesXml, retrieveArtifactsNamesList(mavenDependencyList))
+
+  writeDependenciesVersionsXml(
+    pathToVersionsXml,
+    retrieveArtifactsVersionsList(mavenDependencyList)
+  )
+  val licenseLinkToIndexNameMap = writeDependenciesLicensesXml(
+    pathToLicensesTextsXml,
+    copyrightLicenseSet
+  )
+
+  writeDependenciesLicenseTextsArray(
+    pathToLicenseTextArrayXml,
+    licenseLinkToIndexNameMap,
+    mavenDependencyList
+  )
+
+  writeDependenciesLicenseNamesArray(
+    pathToLicenseNamesArrayXml,
+    copyrightLicenseSet,
+    mavenDependencyList
+  )
+
+  println("Script execution completed.")
+}
+
+private fun retrieveArtifactsNamesList(mavenDependencyList: List<MavenDependency>): List<String> {
+  val artifactNamesList = mutableListOf<String>()
+  mavenDependencyList.forEach { dependency ->
+    artifactNamesList.add(dependency.artifactName)
+  }
+  return artifactNamesList.toList()
+}
+
+private fun retrieveArtifactsVersionsList(
+  mavenDependencyList: List<MavenDependency>
+): List<String> {
+  val artifactVersionsList = mutableListOf<String>()
+  mavenDependencyList.forEach { dependency ->
+    artifactVersionsList.add(dependency.artifactVersion)
+  }
+  return artifactVersionsList.toList()
```

Comment:
More Kotlin idiomatic:

```suggestion
  return mavenDependencyList.map { it.artifactVersion }
```

Ditto elsewhere in this file.

Reply:
Done.